### ISSUE TITLE
About training process

### ISSUE THREAD

**Issue by lucky9-cyou** [Eaphan/NCLR#5]
Thanks for your awesome work. I would like to ask if you have encountered situations where the input tensor is nan during your training. When I started training, it could be stable for a while, but then the input tensor kept becoming nan.
![image](https://github.com/user-attachments/assets/591464f3-8aeb-4ad5-bbd3-e54a3a1931dc)

**Comment by shajnnnbbd** [comment]
I'd like to ask if you've solved this problem yet?

> Thanks for your awesome work. I would like to ask if you have encountered situations where the input tensor is nan during your training. When I started training, it could be stable for a while, but then the input tensor kept becoming nan. ![image](https://private-user-images.githubusercontent.com/56758695/384682845-591464f3-8aeb-4ad5-bbd3-e54a3a1931dc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE0OTA3NDAsIm5iZiI6MTczMTQ5MDQ0MCwicGF0aCI6Ii81Njc1ODY5NS8zODQ2ODI4NDUtNTkxNDY0ZjMtOGFlYi00YWQ1LWJiZDMtZTU0YTNhMTkzMWRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDA5MzQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxYTRiNTZmOGU4NmE4Mzg2N2UyZWUwZTRlOGYwY2Q0ODA3MWVlMjkyOThmZDhkOTBkYmY3YTQ2ZTU0YmI3MjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lNDVj0uCtIdMQt0zYRbJmaKSbpQ9zzhZLXoq4jIAdSs)

**Comment by lucky9-cyou** [comment]
> I'd like to ask if you've solved this problem yet?
> 
> 
> 
> > Thanks for your awesome work. I would like to ask if you have encountered situations where the input tensor is nan during your training. When I started training, it could be stable for a while, but then the input tensor kept becoming nan. ![image](https://private-user-images.githubusercontent.com/56758695/384682845-591464f3-8aeb-4ad5-bbd3-e54a3a1931dc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE0OTA3NDAsIm5iZiI6MTczMTQ5MDQ0MCwicGF0aCI6Ii81Njc1ODY5NS8zODQ2ODI4NDUtNTkxNDY0ZjMtOGFlYi00YWQ1LWJiZDMtZTU0YTNhMTkzMWRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDA5MzQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxYTRiNTZmOGU4NmE4Mzg2N2UyZWUwZTRlOGYwY2Q0ODA3MWVlMjkyOThmZDhkOTBkYmY3YTQ2ZTU0YmI3MjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lNDVj0uCtIdMQt0zYRbJmaKSbpQ9zzhZLXoq4jIAdSs)
> 
> 
> 
> 

No, it's caused by the number of keypoints. The number of kps is zero.

**Comment by shajnnnbbd** [comment]
> > I'd like to ask if you've solved this problem yet?
> > > Thanks for your awesome work. I would like to ask if you have encountered situations where the input tensor is nan during your training. When I started training, it could be stable for a while, but then the input tensor kept becoming nan. ![image](https://private-user-images.githubusercontent.com/56758695/384682845-591464f3-8aeb-4ad5-bbd3-e54a3a1931dc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE0OTA3NDAsIm5iZiI6MTczMTQ5MDQ0MCwicGF0aCI6Ii81Njc1ODY5NS8zODQ2ODI4NDUtNTkxNDY0ZjMtOGFlYi00YWQ1LWJiZDMtZTU0YTNhMTkzMWRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDA5MzQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxYTRiNTZmOGU4NmE4Mzg2N2UyZWUwZTRlOGYwY2Q0ODA3MWVlMjkyOThmZDhkOTBkYmY3YTQ2ZTU0YmI3MjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lNDVj0uCtIdMQt0zYRbJmaKSbpQ9zzhZLXoq4jIAdSs)
> 
> No, it's caused by the number of keypoints. The number of kps is zero.

Thank you for your reply.

**Comment by lucky9-cyou** [comment]
> > > I'd like to ask if you've solved this problem yet?
> > > > Thanks for your awesome work. I would like to ask if you have encountered situations where the input tensor is nan during your training. When I started training, it could be stable for a while, but then the input tensor kept becoming nan. ![image](https://private-user-images.githubusercontent.com/56758695/384682845-591464f3-8aeb-4ad5-bbd3-e54a3a1931dc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE0OTA3NDAsIm5iZiI6MTczMTQ5MDQ0MCwicGF0aCI6Ii81Njc1ODY5NS8zODQ2ODI4NDUtNTkxNDY0ZjMtOGFlYi00YWQ1LWJiZDMtZTU0YTNhMTkzMWRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDA5MzQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxYTRiNTZmOGU4NmE4Mzg2N2UyZWUwZTRlOGYwY2Q0ODA3MWVlMjkyOThmZDhkOTBkYmY3YTQ2ZTU0YmI3MjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lNDVj0uCtIdMQt0zYRbJmaKSbpQ9zzhZLXoq4jIAdSs)
> > 
> > 
> > No, it's caused by the number of keypoints. The number of kps is zero.
> 
> Thank you for your reply.

I tried using 1/9 of the data and training with a single card. It seems to stabilize after a few epochs, and no more NaNs appear.

**Comment by Eaphan** [comment]
Take it easy. It does not affect the training process.

**Comment by Kyekyekye** [comment]
In file "train.py"
after the line  "loss, metrics = model(batch)"
replace all the NAN values in metrics with valid values (0 or 9999 or values from last iteration)

or just ignore it ;p
